### DISCUSSION TITLE
Next app with buildable packages

### DISCUSSION THREAD

**Discussion by bart-krakowski** [nrwl/nx#9910] [General]
Is there any way to run the `next:serve` executor, which is listening to changebenefitslt packages?
My packages are buildable using tsc, so I tried to add `tsc --watch` as `dependsOn`:
app's project.json: 
```
"dev": {
  "executor": "@nrwl/next:server",
  "options": {
    "buildTarget": "app:build",
    "dev": true
  },
  "dependsOn": [
    {
      "target": "dev",
      "projects": "dependencies"
    }
  ],
  "configurations": {
    "production": {
      "buildTarget": "app:build:production",
      "dev": false
    }
  }
},
```

package's project.json:
```
"dev": {
  "executor": "@nrwl/js:tsc",
  // ...
},
```

I know I can use `next-transpile-modules`, but I guess in this way I'm losing the benefits of using `nx affected` on packages.

**Comment by awhiteside1** [comment]
We have the same issue currently. At the moment we have a custom executor that just calls the buildable lib's build command in watch mode while concurrently starting the next serve option. It's a bit temperamental though, so interested in a proper NX solution.
